lib/task_helpers/imports/custom_buttons.rb

```diff
@@ -0,0 +1,82 @@
+require "English"
+
+module TaskHelpers
+  class Imports
+    class CustomButtons
+      class ImportArInstances
+        def self.import(obj_hash)
+          new.import(obj_hash)
+        end
+
+        def import(obj_hash)
+          ActiveRecord::Base.transaction { obj_hash.each { |obj_def| create_object(*obj_def) } }
+        end
+
+        def create_object(class_name, obj_array)
+          klass = class_name.camelize.constantize
+
+          obj_array.collect do |obj|
+            begin
+              klass.create!(obj['attributes'].except('guid')).tap do |new_obj|
+                add_children(obj, new_obj)
+                add_associations(obj, new_obj)
+                check_user(new_obj) if new_obj.kind_of?(CustomButton)
```

Comment:
This logic should be called from `custom_button_post`.  Then you do not need this check for the class type.